### PR TITLE
HAI-2243 New API for getting hakemukset for hanke

### PR DESCRIPTION
**Pull request by umeetiusbaar**
# Description
Move API endpoint `/hankkeet/<hankeid>/hakemukset` from HankeController to new HakemusController.

## Type of change

- [ ] Bug fix 
- [x] New feature 
- [ ] Other

# Instructions for testing
Endpoint `/hankkeet/<hankeid>/hakemukset` should work as before.

# Checklist:

- [x] I have written new tests (if applicable)
- [x] I have ran the tests myself (if applicable)
- [ ] I have made necessary changes to the documentation, link to confluence
 or other location: 

# Other relevant info
Please describe here if there is e.g. some requirements for this change or
 other info that the tester/user needs to know.

### CODE REVIEW

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationController.kt [hunk 1]
```diff
@@ -54,7 +52,26 @@ class ApplicationController(
     private val hankeService: HankeService,
     private val disclosureLogService: DisclosureLogService,
 ) {
-    @GetMapping
+    @GetMapping("/hankkeet/{hankeTunnus}/hakemukset")
+    @Operation(
+        summary = "Get hanke applications",
+        description = "Returns list of applications belonging to a given hanke."
+    )
+    @PreAuthorize("@hankeAuthorizer.authorizeHankeTunnus(#hankeTunnus, 'VIEW')")
```
Comment (corvidian) [inline services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationController.kt:60]: `authorizeHankeTunnus` is a part of the base `Authorizer` class, so it can be called through any authorizer. Using the same authorizer as the other endpoints makes tests simpler.
```suggestion
    @PreAuthorize("@applicationAuthorizer.authorizeHankeTunnus(#hankeTunnus, 'VIEW')")
```

Comment (umeetiusbaar) [reply services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationController.kt]: Good to know! I changed it here and elsewhere where I had used both hankeAuthorizer and applicationAuthorizer.

File: services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/application/ApplicationControllerITest.kt [hunk 2]
```diff
@@ -64,7 +68,7 @@ class ApplicationControllerITest(@Autowired override val mockMvc: MockMvc) : Con
 
     @Autowired private lateinit var applicationService: ApplicationService
     @Autowired private lateinit var hankeService: HankeService
-    @Autowired private lateinit var authorizer: ApplicationAuthorizer
+    @Autowired private lateinit var applicationAuthorizer: ApplicationAuthorizer
```
Comment (corvidian) [inline services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/application/ApplicationControllerITest.kt:71]: Could you revert this change? This causes a lot of noise to the history and it's no longer required.

Comment (umeetiusbaar) [reply services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/application/ApplicationControllerITest.kt]: You're right! Reverted the change and also BASE_URL (well, that could have been better to be the new HAKEMUKSET_BASE_URL but maybe later that change)

**Review by corvidian — APPROVED** [review]
